### PR TITLE
Feature/dmp 417

### PR DESCRIPTION
**Pull request by dianabasilio**
## What is done

[[DMP-417] Loading state cards](https://deacero.atlassian.net/browse/DMP-417)

## How is done

- Added loading prop to  `Card` component that will be used in home and search page for loading the cards.
- Created loadingCard component.

## How to test

1. Run the project via `npm start`
2. You should see the Home Screen. There you will see how the cards are loaded.
3. Type "clavo" to search bar and see the cards being load in search page.

### CODE REVIEW

File: src/components/common/Card/Card.tsx [hunk 1]
```diff
@@ -3,20 +3,25 @@ import styles from "./Card.module.scss";
 interface CardProps {
   title: string;
   size?: "small" | "medium" | "large";
+  loading?: "loading-true" | "loading-false";
```
Comment (markriverac) [inline src/components/common/Card/Card.tsx:6]: `loading` could be a boolean as we are only handling two states

File: src/components/common/Card/Card.tsx [hunk 2]
```diff
@@ -3,20 +3,25 @@ import styles from "./Card.module.scss";
...
   secondaryInfo?: string;
-  imgSrc: string;
+  imgSrc?: string;
 }
 
 const Card: React.FC<CardProps> = ({
   title,
   size = "small",
+  loading = "loading-false",
```
Comment (markriverac) [inline src/components/common/Card/Card.tsx:14]: We set `loading = false` by default

File: src/components/common/Card/Card.tsx [hunk 3]
```diff
@@ -3,20 +3,25 @@ import styles from "./Card.module.scss";
...
 }) => {
   return (
     <div>
-      <div className={`${styles.card} ${styles[size]}`}>
-        <img className={styles.cardImage} src={imgSrc} alt="card" />
+      <div className={`${styles.card} ${styles[size]} ${styles[loading]}`}>
+        {(loading === "loading-false") && (
+          <img className={styles.cardImage} src={imgSrc} alt="card" />
+        )}
+        {loading === "loading-true" && <div className={styles.cardImage} />}
```
Comment (markriverac) [inline src/components/common/Card/Card.tsx:24]: And here we can just
```html
{!loading ? (
          <img className={styles.cardImage} src={imgSrc} alt="card" />
        ) : (
          <div className={styles.cardImage} />
        )}
```

File: src/components/common/Card/Card.tsx [hunk 4]
```diff
@@ -3,20 +3,25 @@ import styles from "./Card.module.scss";
...
   size = "small",
+  loading = "loading-false",
   imgSrc,
   secondaryInfo,
 }) => {
   return (
     <div>
-      <div className={`${styles.card} ${styles[size]}`}>
-        <img className={styles.cardImage} src={imgSrc} alt="card" />
+      <div className={`${styles.card} ${styles[size]} ${styles[loading]}`}>
```
Comment (markriverac) [inline src/components/common/Card/Card.tsx:20]: we create a const for the class based on the state of loading
```javascript
const loadingClass = loading ? "loading" : "";
```
and we use it
``` html
 <div className={`${styles.card} ${styles[size]} ${styles[loadingClass]}`}>
```

File: src/components/common/Card/Card.module.scss [hunk 5]
```diff
@@ -77,3 +78,53 @@
...
+    background: transparent;
+  }
+
+  .cardInfo {
+
+    .cardName {
+      background: transparent;
+    }
+  }
+}
```
Comment (markriverac) [inline src/components/common/Card/Card.module.scss:130]: We don't need these rules

File: src/components/common/Card/Card.module.scss [hunk 6]
```diff
@@ -77,3 +78,53 @@
     text-align: right;
   }
 }
+
+.loading-true {
```
Comment (markriverac) [inline src/components/common/Card/Card.module.scss:82]: Here can we change the name to just `.loading {`

File: src/components/common/Card/Card.module.scss [hunk 7]
```diff
@@ -77,3 +78,53 @@
...
+
+  .cardImage {
+    background: linear-gradient(127deg, #FFFFFF, rgb(219, 214, 214));
+    background-size: 400% 400%;
+    animation: gradient 4s ease infinite;
+  }
+
+  .cardInfo {
```
Comment (markriverac) [inline src/components/common/Card/Card.module.scss:90]: we don't need to call this rule. We can go directly to `.cardName`

**Review by markriverac — CHANGES_REQUESTED** [review]
The animations look really, really nice! 💯 
I just left some comments for code improvement

File: src/components/common/Card/Card.module.scss (hunk 7 above)
Comment (dianabasilio) [reply src/components/common/Card/Card.module.scss:90]: Okey.

**Pull request merged**